go-controller/pkg/node/gateway_udn.go

```diff
@@ -140,8 +146,13 @@ func setBridgeNetworkOfPorts(bridge *bridgeConfiguration, netName string) error
 	return netConfig.setBridgeNetworkOfPortsInternal()
 }
 
-func NewUserDefinedNetworkGateway(netInfo util.NetInfo, networkID int, node *v1.Node, nodeLister listers.NodeLister,
-	kubeInterface kube.Interface, vrfManager *vrfmanager.Controller) *UserDefinedNetworkGateway {
+func NewUserDefinedNetworkGateway(netInfo util.NetInfo, networkID int, node *v1.Node,
+	nodeLister listers.NodeLister, kubeInterface kube.Interface, vrfManager *vrfmanager.Controller,
+	defaultNetworkGateway Gateway) (*UserDefinedNetworkGateway, error) {
```

Comment:
isn't it kind of odd we pass an interface here to assert it is a type gateway struct later, and none of the interface methods:
```
type Gateway interface {
	informer.ServiceAndEndpointsEventHandler
	Init(<-chan struct{}, *sync.WaitGroup) error
	Start()
	GetGatewayBridgeIface() string
	SetDefaultGatewayBridgeMAC(addr net.HardwareAddr)
	Reconcile() error
}
```

are needed by UserDefinedNetworkGateway. It seems like all UserDefinedNetworkGateway wants is a gateway struct so it can access openflowmanager. In that case I think it makes more sense that either just a gateway struct is passed to this function, or the interface is properly extended to expose generic methods and UserDefinedNetworkGateway should use the interface.

Reply:
so this is a great point and something I myself found icky. But in DNCC i.e for default network's gateway code this is what we seem to be doing.. we use the Gateway interface and then assign the gateway struct to it, so in DNCC what I have is interface not the struct. That is why I had to resort to this but I could just straight up extract it as the struct.. lmc what I can do

Reply:
so I tried for extract the `gateway` struct from NCM itself before passing the interface along.. however `gateway` struct is not public outside of the node package only the interface is. That is why I passed the public interface and then extracted the struct from it.

Reply:
I guess we can leave it for now and later refactor it to just take an interface and use interface methods rather than type casting it back to a gateway.